I3S-picking - refactor getting attributes from selected feature

dryabinin-actionengine opened the pull request.

ibgreen reviewed (approved):
This is a lot of code we are adding to the tile layer.

Ultimate we are going to want to merge our changes back to deck.gl.

I would prefer if as much data normalization as possible was done in the I3SLoader so that the TileLayer can be small.

dryabinin-actionengine commented:
> This is a lot of code we are adding to the tile layer.
> 
> Ultimate we are going to want to merge our changes back to deck.gl.
> 
> I would prefer if as much data normalization as possible was done in the I3SLoader so that the TileLayer can be small.

Moved this logic to the `I3SAttributeLoader`. I think make sense to do this code as a part of `I3SAttributeLoader` because we are working with attributes. Also added tests there.

ibgreen commented on modules/i3s/src/i3s-attribute-loader.js:
```diff
@@ -25,3 +26,61 @@ async function parse(data, options) {
+function getFeatureAttributeIndex(layerFeaturesAttributes, featureId) {
+  let featureIndex = -1;
+
+  for (let index = 0; index < layerFeaturesAttributes.length; index++) {
```
When we load the tile, perhaps we could just create a map `attributesByObjectId` in the generated tile so the application can do lookup without calling a function?

dryabinin-actionengine replied:
Will do it in further PR
